packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -4318,21 +4336,29 @@ export class ContainerRuntime
 						!!versions && !!versions[0],
 						0x137 /* "Failed to get version from storage" */,
 					);
-					stats.getVersionDuration = trace.trace().duration;
-					maybeSnapshotTree = await this.storage.getSnapshotTree(versions[0]);
-					assert(!!maybeSnapshotTree, 0x138 /* "Failed to get snapshot from storage" */);
-
-					stats.snapshotVersion = versions[0].id;
+					snapshotTree = await this.storage.getSnapshotTree(versions[0]);
+					assert(!!snapshotTree, 0x138 /* "Failed to get snapshot from storage" */);
+					props.snapshotVersion = versions[0].id;
 				}
 
-				stats.getSnapshotDuration = trace.trace().duration;
-				const latestSnapshotRefSeq = await seqFromTree(maybeSnapshotTree, readAndParseBlob);
-				stats.snapshotRefSeq = latestSnapshotRefSeq;
+				props.getSnapshotDuration = trace.trace().duration;
+				const readAndParseBlob = async <T>(id: string) => readAndParse<T>(this.storage, id);
+				const snapshotRefSeq = await seqFromTree(snapshotTree, readAndParseBlob);
+				props.snapshotRefSeq = snapshotRefSeq;
+				props.newerSnapshotPresent = snapshotRefSeq >= targetRefSeq;
 
-				perfEvent.end(stats);
+				perfEvent.end({ details: props });
+				return snapshotRefSeq;
 			},
 		);
 
+		// If the snapshot that was fetched is older than the target snapshot, return. The summarizer will not be closed
+		// because the snapshot is likely deleted from storage and it so, closing the summarizer will result in the
+		// document being stuck in this state.
+		if (fetchedSnapshotRefSeq < targetRefSeq) {
+			return;
```

Comment:
Should we have some unique telemetry event here to easily find / count these cases?

Reply:
I added a property "newerSnapshotPresent" to "RefreshLatestSummaryAckFetch" event to be able to distinguish this scenario. Adding a new event is also an option but this should be enough  to find and count there cases IMO.